ibis/expr/operations.py

```diff
@@ -2094,21 +2094,20 @@ class Xor(LogicalBinaryOp):
 class Comparison(BinaryOp, BooleanValueOp):
 
     def _maybe_cast_args(self, left, right):
-        if left._can_implicit_cast(right):
-            return left, left._implicit_cast(right)
+        # it might not be necessary?
+        with util.ignoring(com.IbisTypeError):
+            return left, ir.cast(right, left)
 
-        if right._can_implicit_cast(left):
-            return right._implicit_cast(left), right
+        with util.ignoring(com.IbisTypeError):
+            return ir.cast(left, right), right
 
         return left, right
 
     def output_type(self):
-        self._assert_can_compare()
-        return rules.shape_like_args(self.args, 'boolean')
+        if not rules.comparable(self.left, self.right):
+            raise TypeError('Arguments are not comparable')
```

Comment:
Can you show `self.left.type()` and `self.right.type()` in the error message?

Reply:
Sure!